lib/plug/conn.ex

```diff
@@ -26,7 +26,8 @@ defmodule Plug.Conn do
   * `remote_ip` - the IP of the client, example: `{151, 236, 219, 228}`. This field is meant to
     be overwritten by plugs that understand e.g. the `X-Forwarded-For` header or HAProxy's PROXY
     protocol. It defaults to peer's IP.
-  * `req_headers` - the request headers as a list, example: `[{"content-type", "text/plain"}]`
+  * `req_headers` - the request headers as a list, example: `[{"content-type", "text/plain"}]`. 
+    Note all headers will be downcased.
```

Comment:
If you modify the resp_headers map directly no read is happening, and cowboy will ignore the header. 

Maybe we should add a note "Please don't modify this directly, please use put_resp_header/2"